rts/Rendering/Env/Decals/GroundDecalHandler.cpp

```diff
@@ -273,31 +273,32 @@ void CGroundDecalHandler::AddTexturesFromTable()
 
 		// check if loaded for real
 		// can't use atlas->TextureExists() as it's only populated after Finalize()
-		maxUniqueScars += atlasTex->GetAllocator()->contains(mainName);
+		if (atlasTex->GetAllocator()->contains(mainName)) {
+			maxUniqueScars++
+		} else {
+			LOG_L(L_WARNING, "Failed to load decal %s", mainName);
+		}
 	}
 
-	if (maxUniqueScars == scarTblSize)
-		return;
-
-	// fill the gaps in case the loop above failed to load some of the scar textures
-	const std::vector<std::string> scarMainTextures = CFileHandler::FindFiles("bitmaps/scars/", "scar?.*");
-	const size_t scarsExtraNum = scarMainTextures.size();
+	if (maxUniqueScars < scarTblSize) {
```

Comment:
Alternatively we load scars _after_ and return early (same bug-prone logic, not ideal, but reduces nesting), or make separate contained functions for scar and lua decals load (better).

This is the simplest fix for the issue.